[Nicholas Patrick] iP

NicholasPatrick opened the pull request:
# SEMIPRIME

1. Hello
2. there

- General
- Kenobi

```py
>>> 5**5**5**5 % 1000000007
539296137
>>> 1/0
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ZeroDivisionError: division by zero
```

- [X] red
- [ ] white
- [X] lotus

🦷

> baby shark doo doo doo doo doo doo

[I wonder how many people linked the same video](https://youtu.be/dQw4w9WgXcQ)

`p * q = 319726508485703723121005088952685982689147398772004066170145231578269575399229`

**Challenge** to _anyone_ reading this. Find the ~smaller~ prime factor.

Commit:
Add Gradle support

NicholasPatrick committed:
Do Level 1

NicholasPatrick committed:
Do Level 2

NicholasPatrick committed:
Do Level-3 and add method descriptions.

NicholasPatrick committed:
Do Level-4

NicholasPatrick committed:
Do A-TextUiTesting and fix a missed bug from Level-4

NicholasPatrick committed:
Do Level-6 and small fix Level-4. level-5 skipped because no error

NicholasPatrick committed:
Update ui testing to reflect Level-6.

NicholasPatrick committed:
Create SaveFile class for read and write.

NicholasPatrick committed:
Do Level-7

NicholasPatrick committed:
Do Level-8

NicholasPatrick committed:
Merge pull request #1 from NicholasPatrick/branch-Level-7

Branch level 7

NicholasPatrick committed:
Merge branch 'master' into branch-Level-8

NicholasPatrick committed:
Merge pull request #2 from NicholasPatrick/branch-Level-8

Do Level-8

NicholasPatrick committed:
Fix merge conflict resolution errors

NicholasPatrick committed:
Remove left over debugging messages

NicholasPatrick committed:
Create Ui class for input and output

NicholasPatrick committed:
Create TaskList class and move task related activities.

NicholasPatrick committed:
Create Parser class for parsing user inputs.

NicholasPatrick committed:
Do A-Packages

NicholasPatrick committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

NicholasPatrick committed:
Do A-Gradle

NicholasPatrick committed:
Fix A-Gradle

NicholasPatrick committed:
Do A-JUnit

NicholasPatrick committed:
Fix A-JUnit

NicholasPatrick committed:
Do A-JavaDocs

NicholasPatrick committed:
Do A-CodingStandard

NicholasPatrick committed:
Do Level-9

NicholasPatrick committed:
Merge pull request #3 from NicholasPatrick/branch-A-JavaDoc

Do A-JavaDocs

NicholasPatrick committed:
Merge branch 'master' into branch-A-CodingStandard

NicholasPatrick committed:
Merge pull request #4 from NicholasPatrick/branch-A-CodingStandard

Do A-CodingStandard

NicholasPatrick committed:
Merge pull request #5 from NicholasPatrick/branch-Level-9

Do Level-9

NicholasPatrick committed:
Fix bug to update save file after deletion

farisanadia commented on src/main/java/functional/Task.java:
```diff
@@ -0,0 +1,98 @@
+   * @return Returns 'X' for done tasks and ' ' for not done tasks.
+   */
+
+  public char mark() {
```
maybe this can be renamed markTask isntead? 
 _"Names representing methods must be verbs and written in camelCase."_

farisanadia commented on src/main/java/functional/Task.java:
```diff
@@ -0,0 +1,98 @@
+   * Mark a task as not done.
+   */
+
+  public void undo() {
```
maybe this can be renamed undoTask instead?
_" Names representing methods must be verbs and written in camelCase."_

farisanadia commented on src/main/java/technical/TaskList.java:
```diff
@@ -0,0 +1,303 @@
+    StringBuilder deadlineName = new StringBuilder();
+    StringBuilder deadlineDeadlineString = new StringBuilder();
+    boolean byFlagRead = false;
+    for (int i = 1; i < arguments.length; ++i) {
+      if (arguments[i].equals("/by") && !byFlagRead) {
+        byFlagRead = true;
+        continue;
+      }
+      if (byFlagRead) {
+        if (deadlineDeadlineString.length() != 0) {
+          deadlineDeadlineString.append(' ');
+        }
+        deadlineDeadlineString.append(arguments[i]);
+      } else {
+        if (deadlineName.length() != 0) {
+          deadlineName.append(' ');
+        }
+        deadlineName.append(arguments[i]);
+      }
+    }
+    if (deadlineName.length() == 0 || deadlineDeadlineString.length() == 0) {
+      Ui.reply(new String[]{"Format the command as follows:",
+          "deadline <deadline name> /by <deadline>"});
+      return;
+    }
+    LocalDateTime deadlineDeadline;
+    try {
+      deadlineDeadline = LocalDateTime
+          .parse(deadlineDeadlineString.toString(),
+          DateTimeFormatter.ofPattern("yyyy/M/d H:m:s"));
+    } catch (DateTimeParseException e) {
+      Ui.reply("Please format your date as"
+          + " \"year/month/date hour/minute/second\" (24 hour format).");
+      return;
+    }
+    taskList.add(new Deadline(deadlineName.toString(), deadlineDeadline));
+    SaveFile.addData(taskList.get(taskList.size() - 1).toData());
+    justAddedComment();
+  }
+
```
maybe this can be added into abstract classes for better oop?

farisanadia commented on src/main/java/technical/TaskList.java:
```diff
@@ -0,0 +1,303 @@
+   *
+   * @param arguments The command arguments.
+   */
+  public static void mark(String[] arguments) throws IOException {
```
maybe this can be renamed markTask instead?

farisanadia commented on src/main/java/technical/TaskList.java:
```diff
@@ -0,0 +1,303 @@
+   *
+   * @param arguments The command arguments.
+   */
+  public static void unmark(String[] arguments) throws IOException {
```
similarly, maybe this can be renamed unmark task instead?

farisanadia reviewed (commented):
Overall, your code is easy to understand with very clear java docs. well done!

minjiunn commented on src/main/java/technical/SaveFile.java:
```diff
@@ -0,0 +1,104 @@
+public class SaveFile {
+  private static final String SAVE_FILE_NAME = ".data.txt";
+  private static final String TEMPORARY_FILE_NAME = ".data_temp.txt";
+  private static final File saveFile = new File(SAVE_FILE_NAME);
```
since final variables don't change, maybe the naming should follow naming of constants, and should be in uppercase.

minjiunn commented on src/main/java/technical/SaveLine.java:
```diff
@@ -0,0 +1,182 @@
+package technical;
```
good use of OOP, by having the job of saving the input details into a line handled by a separate class :)

minjiunn commented on src/main/java/technical/TaskList.java:
```diff
@@ -0,0 +1,303 @@
+   *
+   * @param arguments The command arguments.
+   */
+  public static void event(String[] arguments) throws IOException {
```
method names should be verbs in camel case, so maybe you can consider naming it addEvent, addTodo, etc

minjiunn reviewed (commented):
good overall compliance to coding standard, and good use of camel case throughout :) just a few things that were maybe overlooked, but other than that goodjob!

NicholasPatrick committed:
do A-CheckStyle

NicholasPatrick committed:
Merge pull request #6 from NicholasPatrick/branch-A-CheckStyle

do A-CheckStyle

NicholasPatrick committed:
Revert "do A-CheckStyle"

NicholasPatrick committed:
Merge pull request #7 from NicholasPatrick/revert-6-branch-A-CheckStyle

Revert "do A-CheckStyle"

NicholasPatrick committed:
Revert "Revert "do A-CheckStyle""

NicholasPatrick committed:
Merge pull request #8 from NicholasPatrick/revert-7-revert-6-branch-A-CheckStyle

Revert "Revert "do A-CheckStyle""

NicholasPatrick committed:
Do Level-10

NicholasPatrick committed:
do A-Assertions and fix file loading

NicholasPatrick committed:
Do A-CodeQuality

previously, the code has bad code quality.

code with bad code quality is bad.

the low quality code is replaced with higher quality code

it's replaced with higher quality code because replacing it with lower
quality code is counter-productive.

methods with over 30 lines are simplified. Some default branch issues
are also fixed.

NicholasPatrick committed:
Merge pull request #10 from NicholasPatrick/branch-A-Assertions

do A-Assertions and fix file loading

NicholasPatrick committed:
Merge branch 'master' into branch-A-CodeQuality

NicholasPatrick committed:
Merge pull request #11 from NicholasPatrick/branch-A-CodeQuality

Do A-CodeQuality

NicholasPatrick committed:
do C-Sort

Some users may want their tasks to be sorted in order of time.

Now users have the sort command that will sort their tasks.

Todos always come first. Deadlines and Events are sorted by their
"essential time"s. The essential time for a Deadline is the deadline
itself, while the essential time for an Event depends on whether the
event is upcoming (start time) or not (end time).

NicholasPatrick committed:
Merge pull request #12 from NicholasPatrick/branch-C-Sort

do C-Sort

NicholasPatrick committed:
Fix bugs

NicholasPatrick committed:
Add Ui.png screenshot

NicholasPatrick committed:
add rudimentary user guide

NicholasPatrick committed:
Do A-UserGuide
